include/LightGBM/dataset_loader.h

```diff
@@ -40,25 +46,37 @@ class DatasetLoader {
                                                         const std::unordered_set<int>& categorical_features);
 
  private:
-  Dataset* LoadFromBinFile(const char* data_filename, const char* bin_filename, int rank, int num_machines, int* num_global_data, std::vector<data_size_t>* used_data_indices);
+  Dataset* LoadFromBinFile(const std::vector<const char*>& data_filename, const char* bin_filename, int rank, int num_machines, int* num_global_data, std::vector<data_size_t>* used_data_indices);
 
-  void SetHeader(const char* filename);
+  void SetHeader(const char* filenames);
```

Comment:
Change `filenames` back to `filename`? Since only the first file is used for `SetHeader`.